src/printer.js

```diff
@@ -556,6 +556,9 @@ function genericPrintNoParens(path, options, print, args) {
         (args && args.expandLastArg) ||
         path.getParentNode().type === "JSXExpressionContainer";
 
+      const printTrailingComma =
+        args && args.expandLastArg && shouldPrintComma(options, "all");
```

Comment:
Not sure I understand the relationship between `expandLastArg` and the bug?

Reply:
We we using `expandLastArg` to add a softline in the end *and* print the trailing comma. Now there's another we want to add a softline in the end (`JSXExpressionContainer`) but not the trailing comma, so they are two differente checks now:
- to add a softline: `expandLastArg` or parent in `JSXExpressionContainer`;
- to add a trailing: `expandLastArg` and the option is enabled

Does that make sense?

Reply:
Wouldn't this be clearer?

```js
const printTrailingComma =
  path.getParentNode().type !== "JSXExpressionContainer" &&
  shouldPrintComma(options, "all");
```

Reply:
I'm not sure.. if `expandLastArg` is true it already says that the arrow function is 1) an argument and 2) the last argument.

Checking only if the parent is not a JSX expression works, but seems brittle. I don't know

Reply:
Question is, is this a fix for JSX or is it more general?

Reply:
Ah, yes. That's a fix for a regression on #3140

Reply:
So it's a fix for arrow functions inside a JSXExpression

Reply:
Which is why I suggested doing:

```
  path.getParentNode().type !== "JSXExpressionContainer" &&
```

Using `expandLastArg` might mean other code will hit this path and be missing a trailing comma.

Reply:
Ok sounds fine.  @vjeux you may want to take a look.